lib/ImageProcessor.js

```diff
@@ -85,13 +88,18 @@ class ImageProcessor {
             resize.bucket = resize.bucket || bucket;
             resize.jpegOptimizer = resize.jpegOptimizer || jpegOptimizer;
 
-            promise = promise.then(() => this.execResizeImage(resize, imageData).then((image) => {
-                this.fileSystem.putObject(image);
-            }));
-            processedImages++;
+            promise = promise
+              .then(() => this.execResizeImage(resize, imageData))
+              .then((image) => this.fileSystem.putObject(image))
+              .then(() => Promise.resolve(++processedImages));
         });
 
-        return promise.then(() => new Promise((resolve) => resolve(processedImages)));
+        if ( removeAfterSuccess ) {
```

Comment:
- Write tests here: https://github.com/ysugimoto/aws-lambda-image/pull/138/files#diff-808fe8ba5b404f2158c475fe7636c35aR51
- Right. I moved to outside `if`.
- I've fixed it, thanks.